client/app/components/queries/QueryEditor/QueryEditorControls.jsx

```diff
@@ -153,6 +156,7 @@ EditorControl.propTypes = {
       onToggle: PropTypes.func,
     }),
   ]),
+  autoLimitCheckboxProps: PropTypes.oneOfType([PropTypes.shape(AutoLimitCheckbox.propTypes)]),
```

Comment:
```suggestion
  autoLimitCheckboxProps: PropTypes.oneOfType([
    PropTypes.bool, // `false` to hide
    PropTypes.shape(AutoLimitCheckbox.propTypes),
  ]),
```

I know that @arikfr asked about this one. Although this is not used for this prop, the `false` value can be used to hide the element (check both the use in the component and the `defaultProps` below).